### PR TITLE
Mute KafkaCruiseControlConfig logs during anomaly detection

### LINKED ISSUE #1880: Verbose logs for KafkaCruiseControlConfig makes debug hard
There are a lots of config logs whenever anomaly detector runs:

```
2022/08/10 23:40:30.815 INFO [KafkaCruiseControlConfig] [AnomalyDetector-2] [kafka-cruise-control] [] KafkaCruiseControlConfig values: 
...
...
```

We should mute them.

### PR DESCRIPTION
**Pull request by CCisGG**
This PR resolves #1880.

@mohitpali FYI. I think this logging is not intentional brought in your PR #1809 . Please correct me if I'm wrong.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public class BrokerSetFileResolver implements BrokerSetResolver {
 
   @Override
   public void configure(Map<String, ?> configs) {
-    KafkaCruiseControlConfig config = new KafkaCruiseControlConfig(configs);
+    KafkaCruiseControlConfig config = new KafkaCruiseControlConfig(configs, false);
```
Comment (efeg) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: Why do we even create a new `KafkaCruiseControlConfig` from the passed parameter. Can we avoid this?

**Review by efeg — CHANGES_REQUESTED** [review]

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java (hunk 1 above)
Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: The passed parameter is a Map. Several lines below it used the KafkaCruiseControlConfig object to create a instance:


`        config.getConfiguredInstance(AnalyzerConfig.BROKER_SET_ASSIGNMENT_POLICY_CLASS_CONFIG, BrokerSetAssignmentPolicy.class);`

So I think it's reasonable to use KafkaCruiseControlConfig here.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java [hunk 2]
```diff
@@ -44,7 +44,7 @@ public class BrokerSetFileResolver implements BrokerSetResolver {
 
   @Override
```
Comment (mohitpali) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:45]: not setting ``doLog`` was NOT intentional.
LGTM.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java (hunk 1 above)
Comment (mohitpali) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: @CCisGG is correct. This was done to load the instance and reuse the ``getConfiguredInstance()`` method. I was trying to emulate ``BrokerCapacityConfigFileResolver`` to keep up with the repo standards. 

However, ``BrokerCapacityConfigFileResolver`` just loaded the json file name from `` _configFile = KafkaCruiseControlUtils.getRequiredConfig(configs, CAPACITY_CONFIG_FILE);``.

``KafkaCruiseControlUtils`` did NOT have a method to get the instance. Hence I created an instance of ``KafkaCruiseControlConfig`` to reuse the ``getConfiguredInstance`` method.

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: Creating a new `KafkaCruiseControlConfig` is expensive and I suspect we can avoid it.
 
-- i.e. we can create the `BrokerSetAssignmentPolicy` instance before creating this class in. For examples of this usage, please search for the use of `parameterConfigOverrides` with `*_OBJECT_CONFIG;` for `getConfiguredInstance`.

Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: @efeg I did a code search for parameterConfigOverrides and getConfiguredInstance, but failed to find an valid example for our case. All the examples are calling the getConfiguredInstance method on an KafkaCruiseControlConfig object. Would you mind provide a concrete example?

Comment (efeg) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java:47]: They are supposed to use KafkaCruiseControlConfig object. The `BrokerSetAssignmentPolicy` object should be created from within `BalancingConstraint` constructor using the parameter of the constructor,  which is a `KafkaCruiseControlConfig`. Then you should update `_brokerSetResolver = config.getConfiguredInstance(AnalyzerConfig.BROKER_SET_RESOLVER_CLASS_CONFIG, BrokerSetResolver.class);` to get the extra parameter (i.e. `Map<String, Object> configOverrides`).

Comment (CCisGG) [reply cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java]: Addressed

**Review by efeg — APPROVED** [review]
Thanks for the updates! LGTM.

**Pull request merged**